### ISSUE TITLE
build the drivers yourself

### ISSUE THREAD

**Issue by ooonea** [eduperez/mwlwifi_LEDE#20]
I want to thank you for your great work, which is very useful. I wanted to build the drivers myself with this last commit. And I tried it yesterday, but the result was to brick the router (Linksys WRT3200ACM) and then restore it from the alternative partition, luckily it exists :)
My problem is that I don't know how to download the 18.06.2 version of OpenWRT as well as the .bin one I used to update the router. In fact if I use the command git clone https://....openwrt and then git checkout v18.06.2 I download the version 18.06.2 at the last revision. I searched the internet for a guide to git, but I couldn't find a solution. Could you help me?

**Comment by ThunderMikey** [comment]
https://downloads.openwrt.org/releases/18.06.2/targets/mvebu/cortexa9/
You can download everything from this link.

**Comment by ooonea** [comment]
openwrt-imagebuilder-18.06.2-mvebu-cortexa9.Linux-x86_64.tar.xz
Is this the environment to build the drivers on 18.06.2?

**Comment by ThunderMikey** [comment]
@lucenera I think the one with SDK in the name is used to compile apps from source.

**Comment by ooonea** [comment]
Ok, I'll try, thanks. I leave it open until I find the solution.

**Comment by eduperez** [comment]
I can only share here _my recipe_ to build the packages:

- `git clone https://git.openwrt.org/openwrt/openwrt.git/ openwrt`
- `cd openwrt`
- `git checkout v18.06.2`
- `scripts/feeds update -a`
- `scripts/feeds install -a`
- _copy "`config.seed`" from OpenWrt's download page as "`.config`"_
- `make defconfig`
- `make -j4 download`
- `make -j6 {tools,toolchain}/install`
- `make -j6 -k`

Hope this helps!

**Comment by ooonea** [comment]
You basically told me everything I need. Another question, if you can be as
kind as you have been so far. When you follow the instructions on the kaloz
page I can't find these two strings in the Makefile:
PKG_VERSION: = 10.3.0.17-20160601 PKG_SOURCE_VERSION: =
4bb95ba1aeccce506a95499b49b9b844ecfae8a.
What should I change exactly? What do you change?

Il lun 25 mar 2019, 14:38 Eduardo Pérez <notifications@github.com> ha
scritto:

**Comment by eduperez** [comment]
In your build environment, there should be a "Makefile" inside the "package/kernel/mwlwifi" folder (if my memory serves me right, I am not in front of my computer right now).

Inside that Makefile, there are a couple of lines, that initialize those variables; the values are probably different, but the names should be the same.

PKG_VERSION is used to give a version to the package, and you are mostly free to put here whatever makes sense to you.

PKG_SOURCE_VERSION is the commit identifier at kaloz's repo that you want to biuld (https://github.com/kaloz/mwlwifi/commits/master).

**Comment by ooonea** [comment]
@eduperez hello, I want to thank you for your valuable help and your work. Thanks to your suggestions I finally managed to compile the @kaloz driver by myself. I didn't close the issue before, because I kept getting errors at the end of the make. The problem was my work environment. This time I used Archlinux and followed the guide on the OpenWRT page (https://openwrt.org/docs/guide-developer/build-system/install-buildsystem) and the driver project page (https://github.com/Káloz/mwlwifi). I currently have the compiled driver (in .ipk) for WRT3200ACM on OpenWRT 18.06.4 official (mwlwifi and kmod) with the last commit (31d9386 on 2 March).
